open-build-service-api/tests/integration.rs

```diff
@@ -167,6 +167,112 @@ async fn test_package_list() {
     assert_eq!(linkinfo.xsrcmd5, branch_xsrcmd5);
 }
 
+#[tokio::test]
+async fn test_source_get() {
+    let test_file = "test";
+    let test_contents = b"some file contents here";
+
+    let mock = start_mock().await;
+    mock.add_project(test_project());
+
+    mock.add_package_revision(
+        &test_project(),
+        test_package_1(),
+        MockRevisionOptions {
+            entries: [(
+                test_file.to_owned(),
+                MockEntry::new_with_contents(SystemTime::UNIX_EPOCH, test_contents.to_vec()),
+            )]
+            .into(),
+            ..Default::default()
+        },
+    );
+
+    let obs = create_authenticated_client(mock);
+
+    // TODO: test empty package
+    let file = obs
+        .project(test_project())
+        .package(test_package_1())
+        .source_file(test_file)
+        .await
+        .unwrap();
+    assert_eq!(&file[..], test_contents);
+}
+
+#[tokio::test]
+async fn test_commits() {
+    let test_file = "test";
+    let test_contents = b"some file contents here";
+    let test_entry = CommitEntry::from_contents(test_file.to_owned(), test_contents);
+
+    let mock = start_mock().await;
+
+    mock.add_project(test_project());
+
+    let obs = create_authenticated_client(mock);
+
+    obs.project(test_project())
+        .package(test_package_1())
+        .create()
+        .await
+        .unwrap();
+
+    let commit_result = obs
+        .project(test_project())
+        .package(test_package_1())
+        .commit(&CommitFileList {
+            entries: vec![CommitEntry::from_contents(
+                test_file.to_owned(),
+                test_contents,
+            )],
+        })
+        .await
+        .unwrap();
+    if let CommitResult::MissingEntries(missing) = commit_result {
+        assert_eq!(missing.entries.len(), 1);
+        assert_eq!(missing.entries[0].name, test_entry.name);
+        assert_eq!(missing.entries[0].md5, test_entry.md5);
+    } else {
+        panic!("Expected missing entries, got {:?}", commit_result);
+    }
+
+    obs.project(test_project())
+        .package(test_package_1())
+        .upload_for_commit(test_file, test_contents.to_vec())
+        .await
+        .unwrap();
+
+    let commit_result = obs
+        .project(test_project())
+        .package(test_package_1())
+        .commit(&CommitFileList {
+            entries: vec![CommitEntry::from_contents(
+                test_file.to_owned(),
+                test_contents,
```

Comment:
better to create the commitfilelist once and re-usre it rather then rebuilding it.

Reply:
Well turns out I *had* created it above and then completely forgot to use it :sweat_smile: